packages/theia-extension/cdtcloud/src/node/compilation-service.test.ts

```diff
@@ -0,0 +1,23 @@
+import { CompilationServiceImpl } from "./compilation-service";
+import "reflect-metadata";
+
+describe("CompilationServiceImpl", () => {
+  let compilationServiceImpl: CompilationServiceImpl;
+
+  beforeEach(async () => {
+    compilationServiceImpl = new CompilationServiceImpl();
+  });
+
+  it("should make 2 requests to server", async () => {
+    try {
+      await compilationServiceImpl.compile(
+        "arduino:avr:mega",
+        "e8665d51-4221-4d3c-b892-a672a2af37a2",
+        "C:/Users/kevin/Documents/Arduino/Light_Project_1"
```

Comment:
Make this reproducible by checking in a fixture we can verify against.

Extract the compilation functionality so we can test it in isolation.

We can move the integration testing up until we can simulate devices.